toolchain/check/impl_lookup.h

```diff
@@ -10,12 +10,27 @@
 
 namespace Carbon::Check {
 
-// Looks up the witness to use for a particular type and interface. Returns the
-// witness, or `InstId::None` if the type is not known to implement the
-// interface.
+// Looks up the witnesses to use for a particular type and a facet type naming a
+// set of interfaces required to be implemented for that type, as well as
+// possible constraints on those interfaces.
+//
+// The return value is one of:
+// - A non-empty InstBlockId value, containing an `ImplWitness` instruction for
+//   each required interface in the `query_facet_type_const_id`. This verifies
+//   the facet type is satisfied for the type in `type_const_id`, and provides a
+//   witness for accessing the impl of each interface.
+//
+// - `InstBlockId::None`, indicating lookup failed for at least one required
+//   interface in the `query_facet_type_const_id`. The facet type is not
+//   satisfied for the type in `type_const_id`.
+//
+// - An error value, indicating lookup failed and a diagonstic has been
```

Comment:
Seems like `None` is saying "lookup failed find the full interface", and error is saying something more is wrong.

Reply:
Yes that's right, an error may even mean that the inputs were errors, which is why it says a diagnostic may have already been produced before.

Reply:
Done